mountinfo/mounted_linux.go

```diff
@@ -0,0 +1,82 @@
+package mountinfo
+
+import (
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+func mountedByOpenat2(path string, st *unix.Stat_t) (bool, error) {
+	// This method works for all mounts (incl. bind mounts),
+	// but requires a recent (v5.6+) linux kernel.
+	var dir, file string
+
+	ftype := st.Mode & unix.S_IFMT
+	switch ftype {
+	case unix.S_IFDIR:
+		dir = path
+		file = ".."
+	default:
+		dir, file = filepath.Split(path)
+	}
+
+	dirfd, err := unix.Open(dir, unix.O_RDONLY|unix.O_PATH, 0)
+	if err != nil {
+		return false, &os.PathError{Op: "open", Path: dir, Err: err}
+	}
+	fd, err := openat2(dirfd, file, &openHow{
+		Flags:   unix.O_RDONLY | unix.O_NOFOLLOW,
```

Comment:
This should also be `O_PATH` not `O_RDONLY`.

Reply:
Awesome, that'll remove these not-so-good-looking special cases.